### PR TITLE
docs: Update Contributing docs regarding dependencies

### PR DESCRIPTION
**Pull request by perlpunk**
Issue: https://progress.opensuse.org/issues/56525

### CODE REVIEW

File: docs/Contributing.asciidoc [hunk 1]
```diff
@@ -325,12 +326,19 @@ workers need to be configured to connect to the main web UI port (add `HOST = ht
...
-* Other dependencies need to be added to `openQA.spec` or `os-autoinst.spec`.
-* Perl dependencies need to be added *additionally* to `cpanfile`.
 * To easily get all necessary dependencies on openSUSE you can install the
   package `openQA-devel`. In other cases one can rely on the `cpanfile` and
   read out the dependencies from the spec file for the rest.
 
+=== Adding dependencies
+
+In openQA, there is a `dependencies.yaml` file including a list of dependencies,
+seperated in groups.  Add a new dependency there and run `make update-deps`.
```
Comment (Martchus) [inline docs/Contributing.asciidoc:336]: a double whitespace

Comment (perlpunk) [reply docs/Contributing.asciidoc]: Ok, fixed, although spaces are not significant in asciidoc, and such spaces can appear if one uses vim to format a number of lines with `gq`.

File: docs/Contributing.asciidoc [hunk 2]
```diff
@@ -130,8 +130,9 @@ other modern web frameworks like Sinatra and that have nice and comprehensive
 documentation available at its http://mojolicio.us[home page].
 
 In addition to Mojolicious and its dependencies, several other CPAN modules are
-required by the openQA package. For a full list of hard dependencies, see the
-file `cpanfile` at the root of the openQA repository.
+required by the openQA package. You can find the dependencies in form of
```
Comment (okurz) [inline docs/Contributing.asciidoc:133]: "the dependencies" is ambiguous as cpanfile does not state *all* dependencies, dependencies.yaml does though, right?

Maybe it's better to delete this block of the section further down

File: docs/Contributing.asciidoc [hunk 3]
```diff
@@ -325,12 +326,19 @@ workers need to be configured to connect to the main web UI port (add `HOST = ht
 * Add 3rd party JavaScript and CSS file to `assets/assetpack.def`. When restarting
   the web server the new/updated files are pulled automatically. Also take care to
   <<Contributing.asciidoc#update-asset-cache,update the asset cache for the openSUSE RPM package>>.
-* Other dependencies need to be added to `openQA.spec` or `os-autoinst.spec`.
-* Perl dependencies need to be added *additionally* to `cpanfile`.
 * To easily get all necessary dependencies on openSUSE you can install the
   package `openQA-devel`. In other cases one can rely on the `cpanfile` and
   read out the dependencies from the spec file for the rest.
 
+=== Adding dependencies
```
Comment (okurz) [inline docs/Contributing.asciidoc:333]: How about "managing" dependencies as this is also about removing/changing

File: docs/Contributing.asciidoc (hunk 2 above)
Comment (perlpunk) [reply docs/Contributing.asciidoc:133]: What do you mean with "delete this block of the section further down"?

**Comment by perlpunk** [comment]
> There is also a description for .circleci/dependencies.txt which IMHO isn't up to date with the current approach anymore

To me it looks like it is uptodate.

File: docs/Contributing.asciidoc (hunk 3 above)
Comment (perlpunk) [reply docs/Contributing.asciidoc]: I restructured it a bit.

File: docs/Contributing.asciidoc (hunk 2 above)
Comment (okurz) [reply docs/Contributing.asciidoc]: Sorry, I meant: "Delete this block in favor of the section further down"

**Pull request merged**